src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/flows/AutoRerouteV2Spec.groovy

```diff
@@ -444,21 +443,39 @@ class AutoRerouteV2Spec extends HealthCheckSpecification {
         and: "Init auto reroute (bring ports down on the dstSwitch)"
         antiflap.portDown(islToBreak.dstSwitch.dpId, islToBreak.dstPort)
 
-        then: "Flows are not rerouted and flows status are 'Down'"
+        then: "Flows are not rerouted and system tries to reroute a flow with transit switch"
         def flowPathMap = [(firstFlow.flowId): firstFlowMainPath, (secondFlow.flowId): secondFlowPath]
-        TimeUnit.SECONDS.sleep(rerouteDelay * 2) // it helps to be sure that the auto-reroute operation is completed
-        Wrappers.wait(WAIT_OFFSET) {
-            assert northbound.getLink(islToBreak).state == FAILED
-            // just to be sure that backup ISL is not failed
-            assert northbound.getLink(islToReroute).state == DISCOVERED
+        Wrappers.wait(WAIT_OFFSET * 2) {
+            def firstFlowHistory = northbound.getFlowHistory(firstFlow.flowId).findAll { it.action == "Flow rerouting" }
+            assert firstFlowHistory.last().histories.find { it.action == REROUTE_FAIL }
+            assert !firstFlowHistory.find { it.taskId =~ (/[^a-z_A-Z0-9-\s]* : retry #1 :/) }
+            def secondFlowHistory = northbound.getFlowHistory(secondFlow.flowId).findAll { it.action == "Flow rerouting" }
+            assert secondFlowHistory.findAll { it.taskId =~ (/[^a-z_A-Z0-9-\s]* : retry #1 :/) }.size() > 1
+            // reroute caused by failed ISL on main and backup paths
+            assert secondFlowHistory.findAll { !(it.taskId =~ (/[^a-z_A-Z0-9-\s]* : retry #1 :/)) }.size() > 1
```

Comment:
can we do a more meaningful search here?

Reply:
refactored
```
assert secondFlowHistory.findAll {
                it.details =~ /Reason: ISL (.*) become INACTIVE because of FAIL TIMEOUT (.*)/
            }.size() == 1
```